### PR TITLE
Imporve metrics text format for FE

### LINKED ISSUE #57: Improve metrics API of FE
Currently FE's metrics API (http://fe_host:fe_port/metrics) return metrics like this:

```
...
# HELP starrocks_fe_job job statistics
# TYPE starrocks_fe_job gauge
starrocks_fe_job{job="load", type="HADOOP", state="UNKNOWN"} 0
# HELP starrocks_fe_job job statistics
# TYPE starrocks_fe_job gauge
starrocks_fe_job{job="load", type="HADOOP", state="PENDING"} 0
# HELP starrocks_fe_job job statistics
# TYPE starrocks_fe_job gauge
starrocks_fe_job{job="load", type="HADOOP", state="ETL"} 0
...
```
And we encountered an exception when using prometheus's [text-parser](https://github.com/prometheus/common/blob/8d1c9f84e3f78cb628a20f8e7be531c508237848/expfmt/text_parse.go#L496) to parse the metrics text:

```
text format parsing error in line 40: second HELP line for metric name "starrocks_fe_job"
```
StarRocks should follow the format definition of prometheus and return metrics like this:

```
...
# HELP starrocks_fe_job job statistics
# TYPE starrocks_fe_job gauge
starrocks_fe_job{job="load", type="HADOOP", state="UNKNOWN"} 0
starrocks_fe_job{job="load", type="HADOOP", state="PENDING"} 0
starrocks_fe_job{job="load", type="HADOOP", state="ETL"} 0
...
```

I'll submit a PR to improve it.

### PR DESCRIPTION
**Pull request by dixingxing0**
Fix #57

### CODE REVIEW

File: fe/fe-core/src/main/java/com/starrocks/metric/PrometheusMetricVisitor.java [hunk 1]
```diff
@@ -155,8 +158,11 @@ public void visitJvm(JvmStats jvmStats) {
     public void visit(@SuppressWarnings("rawtypes") Metric metric) {
         // title
         final String fullName = prefix + "_" + metric.getName();
-        sb.append(HELP).append(fullName).append(" ").append(metric.getDescription()).append("\n");
-        sb.append(TYPE).append(fullName).append(" ").append(metric.getType().name().toLowerCase()).append("\n");
+        if (!metricNames.contains(fullName)) {
```
Comment (kangkaisen) [inline fe/fe-core/src/main/java/com/starrocks/metric/PrometheusMetricVisitor.java:161]: Would better add a comment for why `metricNames` need to de-duplication

**Comment by kangkaisen** [comment]
@dixingxing0  Thanks for your contribution.

File: fe/fe-core/src/main/java/com/starrocks/metric/PrometheusMetricVisitor.java (hunk 1 above)
Comment (dixingxing0) [reply fe/fe-core/src/main/java/com/starrocks/metric/PrometheusMetricVisitor.java:161]: I'll add a comment.

File: fe/fe-core/src/main/java/com/starrocks/metric/StarRocksMetricRegistry.java [hunk 2]
```diff
@@ -23,12 +23,15 @@
...
+import java.util.Collection;
+import java.util.Comparator;
 import java.util.List;
+import java.util.PriorityQueue;
 import java.util.stream.Collectors;
 
 public class StarRocksMetricRegistry {
 
-    private List<Metric> metrics = Lists.newArrayList();
+    private Collection<Metric> paloMetrics = new PriorityQueue<>(Comparator.comparing(Metric::getName));
```
[the remark this answers is not in the document]
Comment (dixingxing0) [reply fe/fe-core/src/main/java/com/starrocks/metric/StarRocksMetricRegistry.java]: I've reverted the variable name

**Comment by imay** [comment]
run starrocks_fe_unittest

**Review by imay — APPROVED** [review]

**Pull request merged**